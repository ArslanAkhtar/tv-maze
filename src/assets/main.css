.v-main{
  background-color: black;
  min-height: 90vh;
}


.movie-summary {
  display: -webkit-box;
  -webkit-line-clamp: 3;
  -webkit-box-orient: vertical;
  overflow: hidden;
}

.genre {
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  justify-content: flex-start;
}

.loading {
  display: flex;
  justify-content: center;
  align-items: center;
  height: 100%;
}

.show-detail {
  display: flex;
  align-items: baseline;
  gap: 5px;
  margin: 10px 0;
}

.loadMore {
  width: 100%;
  display: flex;
  justify-content: center;
}

.no-data-found {
  width: 100%;
  display: flex;
  justify-content: center;
  flex-direction: column;
  align-items: center;
  margin-bottom: 40px;
}

.page-not-found {
  display: flex;
  flex-direction: column;
  align-items: center;
  flex-wrap: wrap;
  justify-content: center;
  height: 100%;
  
}