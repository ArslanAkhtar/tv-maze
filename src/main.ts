import './assets/main.css'

import { createApp } from 'vue'
import App from './App.vue'
import router from './router'
import 'vuetify/styles'
import { createVuetify } from 'vuetify'
import * as components from 'vuetify/components'
import * as directives from 'vuetify/directives'
import { fa } from 'vuetify/iconsets/fa'
import { mdi, aliases } from 'vuetify/iconsets/mdi'

import '@mdi/font/css/materialdesignicons.css'

const app = createApp(App)

app.use(router)

const vuetify = createVuetify({
  components,
  directives,
  theme: {
    defaultTheme: 'dark'
  },
  icons: {
    defaultSet: 'mdi',
    aliases,
    sets: {
      mdi,
      fa
    }
  }
})

app.use(vuetify)

app.mount('#app')
