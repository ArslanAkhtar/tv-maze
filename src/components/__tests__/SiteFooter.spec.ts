import { mount } from "@vue/test-utils";
import Footer from "../SiteFooter.vue";
import { describe, it, expect, vi } from "vitest";
import { vuetify } from "@/main";
describe("Footer Component", () => {
	const wrapper = mount(Footer, {
		global: {
			components: {
				Footer,
			},
		},
		plugins: [vuetify],
	});
	it("renders the copyright text", async () => {
		await wrapper.vm.$nextTick();

		const text = wrapper.text();
		expect(text === "2023 â€” Muhammad Arslan").toBe(true);
	});

	it("renders the social media icons", async () => {
		const icons = wrapper.findAll("v-btn");
		expect(icons.length).toBe(3); // Assuming there are 3 icons
	});

	it("opens external link when icon is clicked", async () => {
		const mockOpen = vi.fn();
		window.open = mockOpen;

		await wrapper.vm.$nextTick();

		const icon = wrapper.find("v-btn");
		await icon.trigger("click");

		expect(mockOpen).toHaveBeenCalledWith(
			"https://www.muhammadarslan.nl/#home",
			"_blank"
		);
	});
	test("renders correctly", () => {
		expect(wrapper.element).toMatchSnapshot();
	});
});
