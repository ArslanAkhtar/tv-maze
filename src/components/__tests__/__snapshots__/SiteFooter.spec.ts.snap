// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Footer Component > renders correctly 1`] = `
<v-footer
  class="text-center d-flex flex-column"
>
  <div>
    
    <v-btn
      class="mx-4"
      icon="mdi-web"
      variant="text"
    />
    <v-btn
      class="mx-4"
      icon="mdi-linkedin"
      variant="text"
    />
    <v-btn
      class="mx-4"
      icon="mdi-github"
      variant="text"
    />
    
  </div>
  <v-divider />
  <div
    class="copyright"
  >
    2023 â€” 
    <strong>
      Muhammad Arslan
    </strong>
  </div>
</v-footer>
`;
